tiledb/test/src/attribute.rs

```diff
@@ -1,78 +1,67 @@
 use serde_json::json;
-use tiledb::array::{attribute::FillData, AttributeData};
 
+use tiledb::array::{attribute::FillData, AttributeData};
+use tiledb::filter_list::FilterListData;
 use tiledb::{fn_typed, Datatype};
 
-use crate::strategy::LifetimeBoundStrategy;
+use crate::*;
+
+fn prop_nullable() -> impl Strategy<Value = bool> {
+    any::<bool>()
+}
+
+fn prop_cell_val_num() -> impl Strategy<Value = Option<u32>> {
+    Just(None)
+}
 
-pub fn arbitrary_name() -> impl Strategy<Value = String> {
-    proptest::string::string_regex("[a-zA-Z0-9_]*")
-        .expect("Error creating attribute name strategy")
-        .prop_filter(
-            "Attribute names may not begin with reserved prefix",
-            |name| !name.starts_with("__"),
-        )
+fn prop_fill<T: Arbitrary>() -> impl Strategy<Value = T> {
+    any::<T>()
 }
 
-fn arbitrary_fill_value<T>() -> impl Strategy<Value = T>
-where
-    T: Arbitrary,
-{
-    proptest::prelude::any::<T>()
+fn prop_filters(datatype: Datatype) -> impl Strategy<Value = FilterListData> {
+    prop_filter_pipeline_for_datatype(datatype)
 }
 
-pub fn arbitrary_for_datatype(
+pub fn prop_attribute_for_datatype(
     datatype: Datatype,
 ) -> impl Strategy<Value = AttributeData> {
-    proptest::prelude::any::<bool>().prop_flat_map(move |nullable| {
-        (
-            Just(nullable),
-            if nullable {
-                proptest::prelude::any::<bool>().bind()
-            } else {
-                Just(false).bind()
-            },
-        )
-            .prop_flat_map(move |(nullable, fill_nullable)| {
-                fn_typed!(datatype, DT, {
-                    (
-                        arbitrary_name(),
-                        Just(nullable),
-                        crate::filter::arbitrary_list_for_datatype(datatype),
-                        arbitrary_fill_value::<DT>(),
-                        Just(fill_nullable),
-                    )
-                        .prop_map(
-                            move |(
-                                name,
-                                nullable,
-                                filters,
-                                fill,
-                                fill_nullable,
-                            )| {
-                                AttributeData {
-                                    name,
-                                    datatype,
-                                    nullability: Some(nullable),
-                                    cell_val_num: None,
-                                    fill: Some(FillData {
-                                        data: json!(fill),
-                                        nullability: Some(fill_nullable),
-                                    }),
-                                    filters,
-                                }
-                            },
-                        )
-                        .bind()
-                })
-            })
+    fn_typed!(datatype, DT, {
```

Comment:
@rroelke This and the removal of the custom lifetime strategy are likely the only things you'll really want to consider when reviewing this PR. Everything else is basically just style changes I made while reading code.

Reply:
Yeah, removing that is fine, now that we no longer need the `&'ctx Context` we get everything we need from `boxed`